cli/internal/oauth/util/util.go

```diff
@@ -0,0 +1,23 @@
+package util
+
+import (
+	"errors"
+	"os/exec"
+	"runtime"
+)
+
+// OpenBrowser opens the specified URL in a browser based on OS.
+func OpenBrowser(url string) (err error) {
+	switch runtime.GOOS {
+	case "linux":
```

Comment:
I suspect WSL is not covered (`xdg-open` will not work on WSL2).
In Docker Desktop, we use `wslview`: https://github.com/docker/pinata/blob/675a1c7d8ae965bb44a0679ff9ff7a108b82b9e0/common/cmd/com.docker.backend/internal/auth/browser/browser.go#L27
(notice the test before to determine if the current environment is a real linux or WSL).

Reply:
More discussion about this [here](https://github.com/docker/cli/pull/5244#discussion_r1682401356).